bot/cogs/weather_cog.py

```diff
@@ -0,0 +1,206 @@
+# Thomas Delvaux
+# 12-16-2020
+
+# API Used:
+# 
+
+import logging
+
+import discord
+import discord.ext.commands as commands
+from bot.messaging.events import Events
+from bot.bot_secrets import BotSecrets
+from bot.consts import Colors
+
+#import requests
+import aiohttp
+import asyncio
+import re
+
+log = logging.getLogger(__name__)
+
+class WeatherCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    def getPageData(self,Lat,Lon,res_json,city):
+        pages = []
+        
+        # FIRST PAGE: Current Conditions
+        page = ''
+
+        # Location
+        #city = res_json.get('name',{})
+        page += f'Location: {city} ({Lat},{Lon})\n'
+
+        # Current Temperature
+        temp = res_json.get('current',{}).get('temp',{})
+        page += f'Temperature: {temp}°F\n'
+
+        # Weather Conditions
+        desc = res_json.get('current',{}).get('weather',{})[0].get('description',{}).title()
+        page += f'Condition:\t{desc}\n\n'
+        
+        # Feels Like
+        feels = res_json.get('current',{}).get('feels_like',{})
+        page += f'Feels Like: {feels}°F\n'
+
+        # Humidity
+        hum = res_json.get('current',{}).get('humidity',{})
+        page += f'Humidity: {hum}%\n'
+
+        # Wind Speed and Direction
+        wind = res_json.get('current',{}).get('wind_speed',{})
+        wind_deg = res_json.get('current',{}).get('wind_deg',{})
+
+        # Convert Wind Degrees to Direction
+        # Per http://snowfence.umn.edu/Components/winddirectionanddegrees.htm
+        dirs = ['N', 'NNE', 'NE', 'ENE', 'E', 'ESE', 'SE', 'SSE', 'S', 'SSW', 'SW', 'WSW', 'W', 'WNW', 'NW', 'NNW']
+        ix = round(float(wind_deg) / (360. / len(dirs)))
+        wind_dir = dirs[ix % len(dirs)]
+
+        page += f'Wind: {wind} mph ({wind_dir})'
+        
+        pages.append(page)
+        
+        # MULTI-PAGE: Forecast
+        for i in range(8):
```

Comment:
where does this 8 come from

Reply:
The JSON forecast goes from 0 (today) to 7 (day 7). I suppose I could count the number of JSON entries in the array in case they change the number of days listed.

Reply:
gotcha that makes sense